common/scala/src/main/scala/whisk/core/database/CouchDbStoreProvider.scala

```diff
@@ -18,16 +18,18 @@
 package whisk.core.database
 
 import akka.actor.ActorSystem
+import scala.collection.mutable
 import spray.json.RootJsonFormat
 import whisk.common.Logging
 import whisk.core.WhiskConfig
 import whisk.spi.Dependencies
-import whisk.spi.SpiFactory
+import whisk.spi.SingletonSpiFactory
 
 /**
  * A CouchDB implementation of ArtifactStoreProvider
  */
 class CouchDbStoreProvider extends ArtifactStoreProvider {
+    private val stores =  mutable.Map [String, CouchDbRestStore[_]]()
```

Comment:
This should be a `TrieMap` to make it threadsafe.

Reply:
replaced this with SpiInstanceCaching (with TrieMap)